File: Java/MissingNumber.java
class MissingNumber {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int expectedSum = n * (n + 1) / 2;  // Sum of first n natural numbers
        int actualSum = 0;

        for (int num : nums) {
            actualSum += num;
        }

        return expectedSum - actualSum;  // The difference is the missing number
    }
}

File: Python/missing_number.py
"""
Problem:
Given an array containing n distinct numbers from 0 to n, find the missing number.

Example:
Input: [3, 0, 1] → Output: 2
"""
def missing_number(nums):
    n = len(nums)
    total = n * (n + 1) // 2
    return total - sum(nums)